### ISSUE TITLE
Support for Schema autocomplete

### ISSUE THREAD

**Issue by sebas5384** [graphql/graphiql#1619]
I'm working on top of what it's already supported on codemirror-graphql and adding some new stuff to support autocomplete, here's some videos of what I have so far: https://twitter.com/sebas5384/status/1284981079636574208?s=20

For now the `graphql` mode is doing an excellent job, though things like multiline comments are not supported so I'll have to add that later so users can document their schema.

Hope to upload the code by the end of this week, maybe as a PR, don't know yet.

I know this project is moving towards Monaco editor, but I guess the autocompletion could be reused.

**Comment by acao** [comment]
yes please do! `codemirror-graphql` is still the default for 90% of graphql tooling and it will stay that way until we can improve on the monaco mode. any improvements will benefit the graphql community foe years to come!

**Comment by acao** [comment]
@sebas5384 also, please note that `monaco-graphql` already has schema autocompletion support

**Comment by acao** [comment]
most of this behavior is implemented at the LSP level, which `monaco-graphql` has implemented already directly.  pretty soon we will have peek definition and symbols support with monaco-graphql, as well, which is simpler because it implements the LSP protocol. currently, `monaco-graphql` supports autocompletion, validation, and other schema features when editing an SDL file by default because the LSP does.

with `codemirror-graphql`, because codemirror does not use LSP protocol, we have to manually implement another layer of business logic to transform `codemirror` event input to interoperate with the LSP (graphql-language-service-interface methods, etc)

this is part of the reason we are moving to `monaco-graphql`, because it more naturally consumes the LSP interface.

**Comment by acao** [comment]
you can also see that the LSP server in this repo ( used by `vscode-graphql`, vim coc graphql, other IDE plugins) are already using the full SDL completion, using the same interfaces that the `monaco-graphql` mode is using.
